abcclassroom/template.py

```diff
@@ -141,14 +191,12 @@ def create_template_dir(config, assignment, mode="fail"):
         )
     else:
         if mode == "fail":
-            print(
-                """Directory {} already exists for this course; re-run with
-                '--mode merge' or '--mode delete', or delete / move directory
-                before re-running""".format(
-                    template_path.relative_to(course_dir)
-                )
+            raise Exception(
```

Comment:
I would raise a specific exception rather than just Exception: https://docs.python.org/3/library/exceptions.html#FileExistsError

Reply:
Then, you can just check that you get the right kind of exception raised when the dir already exists, rather than a regex against the error message. (and I have no idea why that regex is failing on windows!)

Reply:
totally...i can definitely test to see what the specific exception is and catch it. but `match=` will test that the message you expect is actually there - in other words that you catch the exception that you think you are catching.

it's super odd that windows is not happy with this check however???!!!

thank you for this input karen!!

Reply:
ok i figured it out. it was a path issue so i just removed it testing the specific path return. i have a few thoughts on what it was but am out of energy to problem solve anymore. i will work on the exception type later just want to first get this all running.